data/transaction/status.go

```diff
@@ -63,30 +67,44 @@ func (sc *StatusComputer) ComputeStatusWhenInStorageKnowingMiniblock() TxStatus
 // ComputeStatusWhenInStorageNotKnowingMiniblock computes the transaction status when transaction is in current epoch's storage
 // Limitation: in this case, since we do not know the miniblock type, we cannot know if a transaction is actually, "invalid".
 // However, when "dblookupext" indexing is enabled, this function is not used.
-func (sc *StatusComputer) ComputeStatusWhenInStorageNotKnowingMiniblock() TxStatus {
-	if sc.isDestinationMe() || sc.isContractDeploy() {
+func (sc *StatusComputer) ComputeStatusWhenInStorageNotKnowingMiniblock(
+	destinationShard uint32,
+	tx *ApiTransactionResult,
+	) TxStatus {
+	receiver := tx.Tx.GetRcvAddr()
+	transactionData := tx.Data
+
+	if sc.isDestinationMe(destinationShard) || sc.isContractDeploy(receiver,transactionData) {
 		return TxStatusSuccess
 	}
 
 	// At least partially executed (since in source's storage)
 	return TxStatusPending
 }
 
-func (sc *StatusComputer) isMiniblockInvalid() bool {
-	return sc.MiniblockType == block.InvalidBlock
+func (sc *StatusComputer) isMiniblockInvalid(miniblockType block.Type) bool {
+	return miniblockType == block.InvalidBlock
 }
 
-func (sc *StatusComputer) isDestinationMe() bool {
-	return sc.SelfShard == sc.DestinationShard
+func (sc *StatusComputer) isDestinationMe(destinationShard uint32) bool {
+	return sc.SelfShard == destinationShard
 }
 
-func (sc *StatusComputer) isContractDeploy() bool {
-	return core.IsEmptyAddress(sc.Receiver) && len(sc.TransactionData) > 0
+func (sc *StatusComputer) isContractDeploy(receiver []byte, transactionData []byte) bool {
+	return core.IsEmptyAddress(receiver) && len(transactionData) > 0
 }
 
 // SetStatusIfIsRewardReverted will compute and set status for a reverted reward transaction
-func (sc *StatusComputer) SetStatusIfIsRewardReverted(tx *ApiTransactionResult) bool {
-	if sc.MiniblockType != block.RewardsBlock {
+func (sc *StatusComputer) SetStatusIfIsRewardReverted(
+	tx *ApiTransactionResult,
+	miniblockType block.Type,
+	headerNonce uint64,
+	headerHash []byte,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
```

Comment:
check for nils for `tx`, `uint64ByteSliceConverter`, `store`?
Since the method is exported, the code can easily break if someone will pass nil on some of the parameters.

Why not passing the `uint64ByteSliceConverter` and `store` as arguments on the constructor?

Reply:
Done!